src/main/java/org/skriptlang/skript/bukkit/tags/elements/ExprTag.java

```diff
@@ -0,0 +1,132 @@
+package org.skriptlang.skript.bukkit.tags.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Keywords;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.lang.util.SimpleLiteral;
+import ch.njol.util.Kleenean;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Tag;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.tags.TagModule;
+import org.skriptlang.skript.bukkit.tags.TagType;
+import org.skriptlang.skript.bukkit.tags.sources.TagOrigin;
+
+@Name("Tag")
+@Description({
+		"Represents a tag which can be used to classify items, blocks, or entities.",
+		"Tags are composed of a value and an optional namespace: \"minecraft:oak_logs\".",
+		"If you omit the namespace, one will be provided for you, depending on what kind of tag you're using. " +
+		"For example, `tag \"doors\"` will be the tag \"minecraft:doors\", " +
+		"while `paper tag \"doors\"` will be \"paper:doors\".",
+		"`minecraft tag` will search through the vanilla tags, `datapack tag` will search for datapack-provided tags " +
+		"(a  namespace is required here!), `paper tag` will search for Paper's custom tags if you are running Paper, " +
+		"and `custom tag` will look in the \"skript\" namespace for custom tags you've registered.",
+		"You can also filter by tag types using \"item\", \"block\", or \"entity\"."
+})
+@Examples({
+		"minecraft tag \"dirt\" # minecraft:dirt",
+		"paper tag \"doors\" # paper:doors",
+		"tag \"skript:custom_dirt\" # skript:custom_dirt",
+		"skript tag \"dirt\" # skript:dirt",
+		"datapack tag \"dirt\" # minecraft:dirt",
+		"datapack tag \"my_pack:custom_dirt\" # my_pack:custom_dirt",
+		"tag \"minecraft:mineable/pickaxe\" # minecraft:mineable/pickaxe",
+})
```

Comment:
Based on how the pattern of this expression is done, I can see theres an option for the type of tag but none of the examples reflect that.... ie: `minecraft item tag "...`
Having some examples that include the type would be helpful.